### ISSUE TITLE
merge seq tab multiple runs after different trimming parameters

### ISSUE THREAD

**Issue by GaelleQ** [benjjneb/dada2#452]
Hi, 
My question refers to #193. I used the function mergeSequenceTables to merge two sequence tables from different runs. I did not use the same trimming parameters (truncLen) for the two runs. In #193 you suggested that in that case, I might need to truncate to a fixed length first. 

Question: 
Would that be an option to use: 
>seqtab_run1<-seqtab_run1 [,nchar(colnames(seqtab_run1)) %in% seq(419,497)] 

and then use the same criteria for seqtab_run2 for instance? The problem with this option is that I might remove a lot of sequences from one run or the other (see below distributions). 

Any suggestion would be great.
Cheers

Distribution of sequence lengths
Run1
![picture1](https://user-images.githubusercontent.com/38726508/39255973-10e5d9d6-48ae-11e8-82df-fee8646d113b.png)
Run2
![picture2](https://user-images.githubusercontent.com/38726508/39255990-18dbf8fa-48ae-11e8-8d3f-996468c80816.png)

**Comment by fanli-gcb** [comment]
What amplicon are you using? The lower stuff looks like non-specific amplification to me so I'd be cool using the 419-497 range you suggested.

Side note: using different `truncLen` doesn't necessarily make it impossible to merge across runs, but using a different `trimLeft` does.

**Comment by GaelleQ** [comment]
Hi fanli-gcb,
Run 1 and Run 2 were both performed on 16S rRNA gene amplicons of the V1-V3 region from marine algae samples. The only difference is that the samples from Run 1 and those from Run 2 were sent to sequencing in different platforms. 
For run 1 I used:
>seqtab_run1<-seqtab_run1 [,nchar(colnames(seqtab_run1)) %in% seq(419,497)] 

For run 2 I used: 
>seqtab_run2<-seqtab_run2 [,nchar(colnames(seqtab_run1)) %in% seq(428,467)] 

Is that ok or should I have the same length range for merging the tables?
Thanks.

**Comment by benjjneb** [comment]
In order to successfully merge, your amplicons need to cover the same gene region. Right now, yours don't because it looks like you used the same primers but different `trimLeft` parameters (see diagram, [ and ] show where `trimLeft` and `trimRight` were set):

```
PRIMER...[xxxxxxxxxxxxxxxxx]...PRIMER
PRIMER.[xxxxxxxxxxxxxxxxxxxx]..PRIMER
```

So to merge, you need to trim the longer reads to exactly the gene region included in the short-read table. Can you clarify what `trimLeft` parameters you used in each run?

The length selection step you suggested: 

> seqtab_run1<-seqtab_run1 [,nchar(colnames(seqtab_run1)) %in% seq(419,497)]

is totally fine and useful, I agree with @fanli-gcb that it looks like you got some non-specific priming, but that will just remove sequences with lengths outside the range you specify. It won't solve the issue that in your two runs different trimmed gene regions were used.

**Comment by fanli-gcb** [comment]
> Right now, yours don't because it looks like you used the same primers but different trimLeft parameters

Out of curiousity, how did you figure this out? Based on the length distributions being slightly shifted?

**Comment by benjjneb** [comment]
> Based on the length distributions being slightly shifted?

Yep, although looking again now I'm not so sure. But @GaelleQ can clarify what `trimLeft` was used.

**Comment by GaelleQ** [comment]
Hi! Thanks @fanli-gcb and @benjjneb for your inputs. 
I used the same trimLeft parameters for both runs:
Run 1:
>out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, trimLeft=c(21,21), truncLen=c(280,280),
                     maxN=0, maxEE=c(2,5), truncQ=2, rm.phix=TRUE,
                     compress=TRUE, multithread=TRUE) 

Run2:
>out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, trimLeft=c(21,21), truncLen=c(280,250),
                     maxN=0, maxEE=c(2,5), truncQ=2, rm.phix=TRUE,
                     compress=TRUE, multithread=TRUE) 

We are looking at the microbiome associated with marine algae so the non-specific priming can come from the presence of the host.

Regarding the length selection step, 
For run 1 I used:

>seqtab_run1<-seqtab_run1 [,nchar(colnames(seqtab_run1)) %in% seq(419,497)]

For run 2 I used:

>seqtab_run2<-seqtab_run2 [,nchar(colnames(seqtab_run1)) %in% seq(428,467)]

Could you confirm that it won't be a problem for merging the tables or should I select the same length range for both runs?
Many Thanks.

**Comment by benjjneb** [comment]
You should be able to merge just fine, but I would use the same length range for both.

**Comment by GaelleQ** [comment]
Great. Thanks a lot @benjjneb
